LgiHdmiCec/LgiHdmiCec.cpp

```diff
@@ -0,0 +1,1280 @@
+                                                (void *)&param,
+                                                sizeof(param));
+        if (IARM_RESULT_SUCCESS != ret)
+        {
+            LOGERR("%s failed result %d", IARM_BUS_CEC_HOST_SetOSDName, ret);
+        }
+    }
+}
+
+namespace WPEFramework
+{
+    namespace Plugin
+    {
+        SERVICE_REGISTRATION(LgiHdmiCec, 1, 0);
+
+        LgiHdmiCec* LgiHdmiCec::_instance = nullptr;
+
+        static std::atomic<int> libcecInitStatus{0};
+
+        LgiHdmiCec::LgiHdmiCec()
+        :   cecSettingEnabled(false),cecEnableStatus(false),smConnection(nullptr),
+            m_scan_id(0), m_updated(false), m_rescan_in_progress(true), m_system_audio_mode(false)
+        {
+            LgiHdmiCec::_instance = this;
+
+            InitializeIARM();
+            device::Manager::Initialize();
+
+            Register(HDMICEC_METHOD_SET_ENABLED, &LgiHdmiCec::setEnabledWrapper, this);
+            Register(HDMICEC_METHOD_GET_ENABLED, &LgiHdmiCec::getEnabledWrapper, this);
+            Register(HDMICEC_METHOD_GET_CEC_ADDRESSES, &LgiHdmiCec::getCECAddressesWrapper, this);
+            Register(HDMICEC_METHOD_SEND_MESSAGE, &LgiHdmiCec::sendMessageWrapper, this);
+            Register(HDMICEC_METHOD_ENABLE_ONE_TOUCH_VIEW, &LgiHdmiCec::enableOneTouchViewWrapper, this);
+            Register(HDMICEC_METHOD_TRIGGER_ACTION, &LgiHdmiCec::triggerActionWrapper, this);
+            Register(HDMICEC_METHOD_SET_PING_INTERVAL, &LgiHdmiCec::setPingIntervalWrapper, this);
+            Register(HDMICEC_METHOD_GET_CONNECTED_DEVICES, &LgiHdmiCec::getConnectedDevicesWrapper, this);
+            Register(HDMICEC_METHOD_SET_NAME, &LgiHdmiCec::setNameWrapper, this);
+            Register(HDMICEC_METHOD_SET_ONE_TOUCH_VIEW_POLICY, &LgiHdmiCec::setOneTouchViewPolicyWrapper, this);
+
+            physicalAddress = 0x0F0F0F0F;
+
+            logicalAddressDeviceType = "None";
+            logicalAddress = 0xFF;
+
+            loadSettings();
+            if (cecSettingEnabled)
+            {
+                setEnabled(cecSettingEnabled);
+                requestRescanning(getNextScanId());
+            }
+            else
+            {
+                setEnabled(false);
+                Utils::persistJsonSettings (CEC_SETTING_ENABLED_FILE, CEC_SETTING_ENABLED, JsonValue(false));
+            }
+        }
+
+        LgiHdmiCec::~LgiHdmiCec()
+        {
+        }
+
+        void LgiHdmiCec::Deinitialize(PluginHost::IShell* /* service */)
+        {
+            CECDisable();
+            LgiHdmiCec::_instance = nullptr;
+
+            DeinitializeIARM();
+
+        }
+
+        const void LgiHdmiCec::InitializeIARM()
+        {
+            if (Utils::IARM::init())
+            {
+                IARM_Result_t res;
+                IARM_CHECK( IARM_Bus_RegisterEventHandler(IARM_BUS_CECMGR_NAME, IARM_BUS_CECMGR_EVENT_DAEMON_INITIALIZED,cecMgrEventHandler) );
```

Comment:
Utils::Synchro::RegisterLockedIarmEventHandler should probably be used for evet registration (see dt/helpers/UtilsSynchroIarm.hpp) - here and further on (comapre to method rejestrations at DisplaySettings/DisplaySettings.cpp)